webapp-django/crashstats/crashstats/models.py

```diff
@@ -987,11 +987,19 @@ def get(self, bugs):
             url = settings.BZAPI_BASE_URL + (
                 '/bug?id=%(bugs)s&include_fields=%(fields)s' % params
             )
-            response = requests_retry_session().get(
+            response = requests_retry_session(
+                # BZAPI isn't super reliable, so be extra patient
+                retries=5,
+                # 502 = Bad Gateway
+                # 504 = Gateway Time-out
+                status_forcelist=(500, 502, 504),
+            ).get(
                 url,
                 headers=headers,
                 timeout=self.BUGZILLA_REST_TIMEOUT,
             )
+            # Better fail here than getting a JsonDecodeError on the next line.
+            assert response.status_code == 200, response.status_code
```

Comment:
I think I'm more in favor of something like
```python
class BZAPIHttpResponseError(Exception):
    """happens when we get a unexpected response from the bugzilla REST API"""

...

if response.status_code != 200:
    raise BZAPIHttpResponseError(response.status_code)
```
We can tune the name but it'd be a heck of a lot more informative as a traceback.